scripts/workspace-conversion/workspace-to-dataset.py

```diff
@@ -0,0 +1,296 @@
+    else:
+        print("---------------------------------------------------")
+        print(f"There was a problem running this request: ", message)
+        print(json.loads(response.text)["message"])
+        print("\nRequest:")
+        print(f"\t{response.request.url}")
+        print(f"\t{response.request.headers}")
+        print(f"\t{response.request.body}")
+        print("---------------------------------------------------")
+
+
+def getAccessToken():
+    proc = subprocess.Popen(
+        [f"gcloud auth login {user} --brief"], stdout=subprocess.PIPE, shell=True
+    )
+    proc.communicate()
+
+    printProc = subprocess.Popen(
+        ["gcloud auth print-access-token"], stdout=subprocess.PIPE, shell=True
+    )
+    (out, err) = printProc.communicate()
+
+    return out.decode("ASCII").strip()
+
+
+def getWorkspace(accessToken):
+    print("Getting workspace information from rawls...")
+    headers = {
+        "Authorization": f"Bearer {accessToken}",
+        "Content-Type": "application/json",
+    }
+
+    # Get dataset list once in if lowerPolicy == of collision later
+    response = requests.get(
+        f"{urlWorkspace}/{workspaceNamespace}/{workspaceName}", headers=headers
+    )
+    responseData = json.loads(response.text)
+    print("loaded workspace information from rawls")
+    return responseData
+
+
+def mapDatasetReleasePolicy(policyString):
+    lowerPolicy = policyString.lower()
+    if lowerPolicy == "no restrictions" or lowerPolicy == "no restriction":
+        return "TerraCore:NoRestriction"
+    if lowerPolicy == "general research use":
+        return "TerraCore:GeneralResearchUse"
+    if lowerPolicy == "no population origins or ancestry research":
+        return "TerraCore:NPOA"
+    if lowerPolicy == "no general methods research":
+        return "TerraCore:NMDS"
+    if lowerPolicy == "genetic studies only":
+        return "TerraCore:GSO"
+    if lowerPolicy == "clinical care use":
+        return "TerraCore:CC"
+    if lowerPolicy == "publication required":
+        return "TerraCore:PUB"
+    if lowerPolicy == "collaboration required":
+        return "TerraCore:COL"
+    if lowerPolicy == "ethics approval required":
+        return "TerraCore:IRB"
+    if lowerPolicy == "geographical restriction":
+        return "TerraCore:GS"
+    if lowerPolicy == "publication moratorium":
+        return "TerraCore:MOR"
+    if lowerPolicy == "return to database/resource":
+        return "TerraCore:RT"
+    if lowerPolicy == "non commercial use only":
+        return "TerraCore:NCU"
+    if (
+        lowerPolicy == "not-for-profit use only"
+        or lowerPolicy == "not for profit use only"
+    ):
+        return "TerraCore:NPC"
+    if lowerPolicy == "not-for-profit, non-commercial use ony":
+        return "TerraCore:NPC2"
+    return policyString
+
+
+def mapDataModality(modalityArray):
+    ret = []
+    for modality in modalityArray:
+        if modality == "Epigenomic":
```

Comment:
I'm not a python programmer, but can this be done using a dictionary instead of an `if`?

Reply:
Ah, updated to this :D